scripts/qa_crowbarsetup.sh

```diff
@@ -1562,6 +1572,32 @@ function set_node_role_and_platform()
     rm -f $t
 }
 
+# set a node's role
+# must be run after discovery and before allocation
+function set_node_role()
+{
+    node="$1"
+    role="$2"
+    local t=$(mktemp).json
+    knife node show -F json "$node" > $t
+    json-edit $t -a normal.crowbar_wall.intended_role -v "$role"
+    knife node from file $t
+    rm -f $t
+}
+
+# set a node's platform
+# must be run after discovery and before allocation
+function set_node_platform()
+{
+    node="$1"
+    platform="$2"
+    local t=$(mktemp).json
+    knife node show -F json "$node" > $t
+    json-edit $t -a normal.target_platform -v "$platform"
```

Comment:
I dislike the code-duplication. IMHO there should be one function that gets a 3rd param that is either normal.target_platform or normal.crowbar_wall.intended_role